components/TextFields/tests/unit/ContainedInputView/MDCBaseTextFieldTests.m

```diff
@@ -14,89 +14,527 @@
+
+  // When
   UIView *sideView = [self createSideView];
   textField.trailingView = sideView;
+
+  // Then
   XCTAssertTrue(textField.leftView == textField.trailingView,
                 @"The trailing view should be the left view.");
 }
 
-- (void)testLeadingViewModeLTR {
+- (void)testLeadingViewModeEqualsLeftViewModeInLTR {
+  // Given
   MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+
+  // When
   textField.layoutDirection = UIUserInterfaceLayoutDirectionLeftToRight;
   textField.leadingViewMode = UITextFieldViewModeAlways;
+
+  // Then
   XCTAssertTrue(textField.leftViewMode == textField.leadingViewMode,
                 @"The leading view mode should be equal to the left view mode.");
 }
 
-- (void)testLeadingViewModeRTL {
+- (void)testLeadingViewModeEqualsRightViewModeInRTL {
+  // Given
   MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+
+  // When
   textField.layoutDirection = UIUserInterfaceLayoutDirectionRightToLeft;
   textField.leadingViewMode = UITextFieldViewModeAlways;
+
+  // Then
   XCTAssertTrue(textField.rightViewMode == textField.leadingViewMode,
                 @"The leading view mode should be equal to the right view mode.");
 }
 
-- (void)testTrailingViewModeLTR {
+- (void)testTrailingViewModeEqualsRightViewModeInLTR {
+  // Given
   MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+
+  // When
   textField.layoutDirection = UIUserInterfaceLayoutDirectionLeftToRight;
   textField.trailingViewMode = UITextFieldViewModeAlways;
-  XCTAssertTrue(textField.rightView == textField.trailingView,
+
+  // Then
+  XCTAssertTrue(textField.rightViewMode == textField.trailingViewMode,
                 @"The trailing view mode should be equal to the right view mode.");
 }
 
-- (void)testTrailingViewModeRTL {
+- (void)testTrailingViewModeEqualsLeftViewModeInRTL {
+  // Given
   MDCBaseTextField *textField = [[MDCBaseTextField alloc] initWithFrame:CGRectMake(0, 0, 100, 60)];
+
+  // When
   textField.layoutDirection = UIUserInterfaceLayoutDirectionRightToLeft;
   textField.trailingViewMode = UITextFieldViewModeAlways;
+
+  // Then
   XCTAssertTrue(textField.leftViewMode == textField.trailingViewMode,
                 @"The trailing view mode should be equal to left view mode.");
 }
 
+- (void)testAdjustTextAreaFrameWithParentClassTextAreaFrame {
+  // Given
+  CGRect desiredTextAreaFrame = CGRectMake(30, 50, 120, 20);
+  CGRect pretendParentClassTextAreaFrame = CGRectMake(20, 30, 120, 50);
+  CGFloat pretendSystemDefinedHeight = CGRectGetHeight(pretendParentClassTextAreaFrame);
+  CGFloat desiredTextAreaMidY = CGRectGetMidY(desiredTextAreaFrame);
+  CGFloat halfOfPretendSystemDefinedHeight = (pretendSystemDefinedHeight * (CGFloat)0.5);
+  CGFloat desiredTextAreaMinY = desiredTextAreaMidY - halfOfPretendSystemDefinedHeight;
```

Comment:
tip: generally speaking you don't need to define a variable for each mathematical expression. E.g. consider simply using `(pretendSystemDefinedHeight * (CGFloat)0.5)` here instead.